mm2src/mm2_main/src/lp_network.rs

```diff
@@ -155,10 +158,18 @@ async fn process_p2p_message(
             Some(lp_swap::TX_HELPER_PREFIX) => {
                 if let Some(pair) = split.next() {
                     if let Ok(Some(coin)) = lp_coinfind(&ctx, pair).await {
-                        match coin.send_raw_tx_bytes(&message.data).compat().await {
-                            Ok(id) => log::debug!("Transaction broadcasted successfully: {:?} ", id),
-                            Err(e) => log::error!("Broadcast transaction failed. {}", e),
-                        }
+                        if let Err(e) = coin.tx_enum_from_bytes(&message.data) {
+                            log::error!("Message cannot continue the process due to: {:?}", e);
+                            continue;
+                        };
+
+                        let fut = coin.send_raw_tx_bytes(&message.data);
+                        spawn(async {
+                            match fut.compat().await {
+                                Ok(id) => log::debug!("Transaction broadcasted successfully: {:?} ", id),
+                                Err(e) => log::error!("Broadcast transaction failed. {}", e),
```

Comment:
If the transaction is broadcasted successfully by 1 of the seednodes then this error will show in all the other seednodes when it's not really an error since the transaction is already broadcasted. We can check for the returned error if it's `"error": { "code": -25, "message": "bad-txns-inputs-missingorspent" }` then we can debug this as `Transaction broadcasted successfully by another seednode`, otherwise show an error.

For more info on `"error": { "code": -25, "message": "bad-txns-inputs-missingorspent" }` you can check
https://github.com/bitcoin/bitcoin/blob/623745ca74cf3f54b474dac106f5802b7929503f/src/rpc/protocol.h#L46
https://github.com/bitcoin/bitcoin/blob/c5ba1d92b633e703c25121c0d73dc0e04baae01b/src/rpc/rawtransaction.cpp#L558
https://github.com/bitcoin/bitcoin/blob/59ac8bacd573891d84e8ab96ff43ea12bd266f26/src/consensus/tx_verify.cpp#L172

I am pretty sure this is the right error to check for, but please try to rebroadcast a transaction that is already in the blockchain or maybe one that were just inserted to the mempool from the latest transactions here https://mempool.space/ as this better mimics our case to check for what error will be returned.

Reply:
Do you think error handling by string comparison is a good idea ? :) And will all the rpc's return exactly same error string? I think we should keep it this way since it is actually an error returned from the rpc server. If the transaction is missing or spent, then we want to see it as an error log on all the seednodes if I am not mistaken. These are my thoughts. Please let me know if I should handle it that way. :)

Reply:
> Do you think error handling by string comparison is a good idea ? :)

No, I don't think this is a good idea, I think the error return type can be refactored but there is another PR open for error refactors that is fixing the error return type of this function, so maybe we can write a todo here for the future since this is a non-blocker I guess.

> And will all the rpc's return exactly same error string?

Yes, I believe so for all the major UTXOs at least, the downside if the rpcs didn't return the same error is that we get the same message we have now which is `Broadcast transaction failed`

> If the transaction is missing or spent, then we want to see it as an error log on all the seednodes if I am not mistaken. These are my thoughts. Please let me know if I should handle it that way. :)

Maybe we can write a more detailed message saying `Broadcast transaction failed. If this transaction is already broadcasted to the blockchain by another seednode, this error can be ignored.` as a temporary solution for now.

Reply:
> Yes, I believe so for all the major UTXOs at least, the downside if the rpcs didn't return the same error, is that we get the same message we have now which is Broadcast transaction failed

But not only Utxo errors we will face with here.

Updated the error log message. :)